### PR TITLE
gatherer: allow same package names in different source files

### LINKED ISSUE #24: "proto package names should not be mixed in the same directory" false positive
### 💥 Sup?

On [this branch](https://github.com/lyft/protoc-gen-star/compare/master...srenatus:sr/bug/pkg-name-clash), running the example protoc plugin in `testdata/protos` yields what I believe to be **a false positive package name clash**:

```console
$ env DEBUG=1 protoc -I . --example_out="plugins:../generated" bug/api/config/config.proto
registering plugin: jsonify
registering module:  printer
reading input
parsing input proto
parsing command-line params
evaluating multi-package mode
single package run
registering plugin: gatherer
initializing plugins
preparing official generator
generating official PGG PBs and gathering PG* AST
[jsonify] Initializing
[gatherer] Initializing
[gatherer][package:config] importPath: "github.com/lyft/protoc-gen-star/testdata/protos/bug/pkg/config"
[gatherer][file:bug/pkg/config/main.proto] f.GetPackage() = lyft.pgs.bar.config
[gatherer][file:bug/pkg/config/main.proto] pkg.ProtoName() = lyft.pgs.bar.config
[gatherer][package:config] importPath: "github.com/lyft/protoc-gen-star/testdata/protos/bug/pkg/config"
[gatherer][file:bug/pkg/config/secondary.proto] f.GetPackage() = lyft.pgs.bar.config
[gatherer][file:bug/pkg/config/secondary.proto] pkg.ProtoName() = lyft.pgs.bar.config
[jsonify] [bug/api/config/config.proto] [Fizz] implementing json.Marshaler/Unmarshaler interface
[gatherer][package:config1] importPath: "github.com/lyft/protoc-gen-star/testdata/protos/bug/api/config"
[gatherer][file:bug/api/config/config.proto] f.GetPackage() = lyft.pgs.api.config
[gatherer][file:bug/api/config/config.proto] pkg.ProtoName() = lyft.pgs.bar.config
2018/06/27 16:31:09 protoc-gen-go: error:[gatherer][file:bug/api/config/config.proto]proto package names should not be mixed in the same directory (lyft.pgs.bar.config, lyft.pgs.api.config)
--example_out: protoc-gen-example: Plugin failed with status code 1.
```

I don't believe the situation set up there is invalid. 😃 

ℹ️ This could be related to the changes done for protobuf v1.1.0, I haven't seen it before. Also, I believe parts of the relevant code paths have been touched recently.

### PR DESCRIPTION
**Pull request by glerchundi**
Fixes #24

### CODE REVIEW

File: gatherer.go [hunk 1]
```diff
@@ -45,6 +45,10 @@ func (g *gatherer) Generate(f *generator.FileDescriptor) {
 	}
 
 	pkg := g.hydratePackage(f, comments)
+	if f.GetPackage() != pkg.ProtoName().String() {
+		g.Debugf("proto package names should not be mixed in the same directory (%s, %s)", f.GetPackage(), pkg.ProtoName())
```
Comment (rodaine) [inline gatherer.go]: This should still be a `g.Assert` like the previous check, please.

**Comment by glerchundi** [comment]
@rodaine sorry for the delay.

The change from `Assert` to `Debug` was deliberetely done, in order to don't fail/stop when that occurs. Take these files as a reference:

`google/api/annotations.proto`:
```
syntax = "proto3";

package google.api;

option go_package = "github.com/salto/annotations-go;annotations";

...
```

`salto/api/annotations.proto`:
```
syntax = "proto3";

package salto.api;

option go_package = "github.com/salto/annotations-go;annotations";

...
```

`myservice.proto`:
```
syntax = "proto3";

package myservice;

import "google/api/annotations.proto";
import "salto/api/annotations.proto";

service MyService {
...
}
```

If we call `protoc` in this way: `protoc -I . --whatever_my_plugin_is_out=. myservice.proto` it seems to call `Generate` three times, one per the legitimate source file: `myservice.proto` and another two for the imported (but not required to generate) files: `{google,salto}/api/annotations.proto`.

The changes recently commited to the repo fixes a bug where `go_option` wasn't respected correctly but with that fix, another one have arised another one with 'false' conflicting packages.

Will fix the build as soon as possible.

**Comment by rodaine** [comment]
See #28, going to be tackling this in another way.

**Comment by glerchundi** [comment]
@rodaine perfect, will close this then, thanks!

**Pull request closed without merging**